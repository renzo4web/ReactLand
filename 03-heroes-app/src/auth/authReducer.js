import { types } from '../types/types';

const authReducer = (state = {}, action) => {
  const { type, payload } = action;

  switch (type) {
    case types.login:
      return { ...state, ...payload, logged: true };

    case types.logout:
      return { ...payload, logged: false };
    case 'default':
      return state;

    default:
      throw new Error(
        `authReducer receibed invalid type ${type}`
      );
  }
};

export { authReducer };
