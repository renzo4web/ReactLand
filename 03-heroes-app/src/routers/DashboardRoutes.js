import { Box } from '@chakra-ui/react';
import React, { useEffect } from 'react';
import {
  Redirect,
  Route,
  Switch,
  useLocation,
} from 'react-router-dom';

import DcScreen from '../components/dc/DcScreen';
import AllHeroesScreen from '../components/heroes/AllHeroesScreen';
import HeroScreen from '../components/heroes/HeroScreen';
import MarvelScreen from '../components/marvel/MarvelScreen';
import SearchScreen from '../components/search/SearchScreen';
import { Navbar } from '../components/ui/Navbar';

const DashboardRoutes = () => {
  const { pathname } = useLocation();
  useEffect(() => {
    const title = `${pathname.replace('/', '')}`;
    document.title = !title
      ? 'All Heroes'
      : `${title.charAt(0).toUpperCase()}${title.slice(
          1
        )} Heroes`;
  }, [pathname]);

  return (
    <Box w="100vw">
      <Navbar />

      <Box>
        <Switch>
          <Route exact path="/marvel">
            <MarvelScreen />
          </Route>
          <Route exact path="/dc">
            <DcScreen />
          </Route>
          <Route exact path="/heroe/:heroeId">
            <HeroScreen />
          </Route>

          <Route path="/search">
            <SearchScreen />
          </Route>

          <Route path="/">
            <AllHeroesScreen />
          </Route>
          <Redirect to="/marvel" />
        </Switch>
      </Box>
    </Box>
  );
};

export default DashboardRoutes;
