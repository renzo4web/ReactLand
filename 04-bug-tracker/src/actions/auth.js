import Swal from 'sweetalert2';

import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { types } from '../types/types';
import { bugsLogout } from './bugs';
import { finishLoading, startLoading } from './ui';

export const startLoginEmailPassword = (email, password) => {
    return (dispatch) => {
        dispatch(startLoading());

        firebase
            .auth()
            .signInWithEmailAndPassword(email, password)
            .then(({ user }) => {
                dispatch(login(user.uid, user.displayName));
                dispatch(finishLoading());
            })
            .catch((err) => {
                dispatch(finishLoading());
                Swal.fire('Error', err.message);
            });
    };
};

export const startGoogleLogin = () => {
    return (dispatch) => {
        firebase
            .auth()
            .signInWithPopup(googleAuthProvider)
            .then(({ user }) => {
                const { displayName, uid } = user;
                dispatch(login(uid, displayName));
            });
    };
};

export const startRegisterWithEmailPasswordName = (name, email, password) => {
    return (dispatch) => {
        firebase
            .auth()
            .createUserWithEmailAndPassword(email, password)
            .then(async ({ user }) => {
                await user.updateProfile({
                    displayName: name,
                });

                dispatch(login(user.id, user.displayName));
            })
            .catch((err) => {
                Swal.fire('Error', err.message);
            });
    };
};

export const login = (uid, displayName) => ({
    type: types.login,
    payload: {
        uid,
        displayName,
    },
});

export const startLogout = () => {
    return async (dispatch) => {
        await firebase.auth().signOut();
        dispatch(bugsLogout());
        dispatch(logout());
    };
};

export const logout = () => ({
    type: types.logout,
});
