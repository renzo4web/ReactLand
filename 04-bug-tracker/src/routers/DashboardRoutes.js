import React from 'react';
import { Switch, Route } from 'react-router-dom';
import BugScreen from '../components/bugTracker/views/BugScreen';
import BugTrackerScreen from '../components/bugTracker/views/Dashboard/BugTrackerScreen';
import SubmitBugScreen from '../components/bugTracker/views/SubmitBugScreen';

const DashboardRoutes = () => {
    return (
        <>
            <Switch>
                <Route exact path='/' component={BugTrackerScreen} />
                <Route exact path='/submit' component={SubmitBugScreen} />
                <Route exact path='/bug/:bugId' component={BugScreen} />
            </Switch>
        </>
    );
};

export default DashboardRoutes;
