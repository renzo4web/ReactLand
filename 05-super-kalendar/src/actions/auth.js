import { types } from '../types/types';
import { fetchWithoutToken, fetchWithToken } from '../helpers/fetch';
import toast from 'react-hot-toast';
import { eventsCleanAll } from './events';

export const startLogin = (email, password) => {
	return async dispatch => {
		// http://localhost:4000/api/auth
		const response = await fetchWithoutToken(
			'auth',
			{ email, password },
			'POST'
		);
		const body = await response.json();

		if (body.ok) {
			localStorage.setItem('token', body.token);
			// keep track initial date of the token created
			localStorage.setItem('token-init-date', new Date().getTime());

			const { uid, name } = body;

			dispatch(
				login({
					uid,
					name,
				})
			);
		} else {
			toast.error(body.msg);
		}
	};
};

export const startRegister = (name, email, password) => {
	return async dispatch => {
		const response = await fetchWithoutToken(
			'auth/new',
			{ email, password, name },
			'POST'
		);

		const body = await response.json();

		if (body.ok) {
			localStorage.setItem('token', body.token);
			// keep track initial date of the token created
			localStorage.setItem('token-init-date', new Date().getTime());

			const { uid, name } = body;

			dispatch(
				login({
					uid,
					name,
				})
			);
		} else {
			toast.error(body.msg);
		}
	};
};

export const startChecking = () => {
	return async dispatch => {
		const response = await fetchWithToken('auth/renew');
		const body = await response.json();

		if (body.ok) {
			localStorage.setItem('token', body.token);
			// keep track initial date of the token created
			localStorage.setItem('token-init-date', new Date().getTime());

			const { uid, name } = body;

			dispatch(
				login({
					uid,
					name,
				})
			);
		} else {
			dispatch(checkingFinish());
		}
	};
};

const checkingFinish = () => ({
	type: types.authCheckingFinish,
});

export const startLogout = () => {
	return dispatch => {
		localStorage.clear();
		dispatch(logout());
		dispatch(eventsCleanAll());
	};
};

const logout = () => ({
	type: types.authLogout,
});

const login = user => ({
	type: types.authLogin,
	payload: user,
});
