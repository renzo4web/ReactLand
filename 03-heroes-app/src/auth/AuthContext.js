import React, {
  createContext,
  useReducer,
  useContext,
  useEffect,
} from 'react';
import { authReducer } from './authReducer';

const Context = createContext();

const init = () => {
  return (
    JSON.parse(localStorage.getItem('user')) || {
      logged: false,
    }
  );
};

const AuthContext = ({ children }) => {
  const value = useReducer(authReducer, {}, init);
  const [auth, dispatch] = value;

  useEffect(() => {
    localStorage.setItem('user', JSON.stringify(auth));
  }, [auth]);

  return (
    <Context.Provider value={value}>
      {children}
    </Context.Provider>
  );
};

export const useAuthContext = () => {
  return useContext(Context);
};

export { AuthContext };
