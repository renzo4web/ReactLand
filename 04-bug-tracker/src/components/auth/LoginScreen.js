import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import validator from 'validator';
import { Link } from 'react-router-dom';
import { startGoogleLogin, startLoginEmailPassword } from '../../actions/auth';
import { useForm } from '../../hooks/useForm';
import { removeError, setError } from '../../actions/ui';

const LoginScreen = () => {
    const initialForm = {
        email: '',
        password: '',
    };
    const { msgError, loading } = useSelector((state) => state.ui);
    const dispatch = useDispatch();
    const [formValues, handleInputChange] = useForm(initialForm);

    const { email, password } = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        if (isFormValid()) {
            dispatch(startLoginEmailPassword(email, password));
        }
    };

    const handleGoogleLogin = () => {
        dispatch(startGoogleLogin());
    };

    const isFormValid = () => {
        if (!validator.isEmail(email)) {
            dispatch(setError('Email is not valid'));
            return false;
        }

        if (password.length < 5) {
            dispatch(
                setError(
                    'Password should be at least 5 characters long and match each other'
                )
            );
            return false;
        }

        dispatch(removeError());
        return true;
    };

    return (
        <>
            <h1>Login</h1>

            {msgError && (
                <strong className='ui__alert-input'>{msgError}</strong>
            )}
            <form onSubmit={handleLogin}>
                <div className='ui__input-group'>
                    <label htmlFor='email'>Email</label>
                    <input
                        value={email}
                        placeholder='test@gmail.com'
                        onChange={handleInputChange}
                        name='email'
                        id='email'
                        type='text'
                    />
                </div>

                <div className='ui__input-group'>
                    <label htmlFor='password'>Password</label>
                    <input
                        value={password}
                        onChange={handleInputChange}
                        name='password'
                        id='password'
                        type='password'
                    />
                </div>

                <button disabled={loading} type='submit' className='auth__btn'>
                    Login
                </button>

                <hr />

                <div>
                    <p>Login with social networks</p>

                    <div onClick={handleGoogleLogin} className='google-btn'>
                        <div className='google-icon-wrapper'>
                            <img
                                className='google-icon'
                                src='https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg'
                                alt='google button'
                            />
                        </div>
                        <p className='btn-text'>
                            <b>Sign in with google</b>
                        </p>
                    </div>
                </div>

                <Link to='/auth/register'>Create new account</Link>
            </form>
        </>
    );
};

export default LoginScreen;
