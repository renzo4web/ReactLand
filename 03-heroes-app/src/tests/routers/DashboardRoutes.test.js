import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import DashboardRoutes from '../../routers/DashboardRoutes';

describe('<DashboardRoutes /> Test', () => {
  test('should render properly', () => {
    const wrapper = mount(
      <AuthContext>
        <MemoryRouter>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
