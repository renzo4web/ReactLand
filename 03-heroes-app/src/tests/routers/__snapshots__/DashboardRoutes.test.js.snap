// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardRoutes /> Test should render properly 1`] = `
<div
  className="css-306mec"
>
  <nav
    className="css-qluv0w"
  >
    <div
      className="css-dsyim5"
      onClick={[Function]}
    >
      <svg
        fill="white"
        viewBox="0 0 20 20"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Menu
        </title>
        <path
          d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"
        />
      </svg>
    </div>
    <div
      className="css-8olkqp"
    >
      <div
        className="chakra-stack css-1agnwj4"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <p
            className="chakra-text css-6vcm9r"
          >
            All
          </p>
        </a>
        <a
          href="/dc"
          onClick={[Function]}
        >
          <p
            className="chakra-text css-6vcm9r"
          >
            DC
          </p>
        </a>
        <a
          href="/marvel"
          onClick={[Function]}
        >
          <p
            className="chakra-text css-6vcm9r"
          >
            Marvel
          </p>
        </a>
        <button
          className="chakra-button css-1xwdflt"
          onClick={[Function]}
          type="button"
        >
          Logout
        </button>
        <form
          onSubmit={[Function]}
        >
          <div
            className="chakra-input__group css-1fzfw9r"
          >
            <div
              className="chakra-input__left-element css-1u2k0bl"
            >
              <svg
                className="chakra-icon css-1xzgr6b"
                focusable={false}
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <input
              className="chakra-input css-1qcmu60"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search your favorite superhero"
              type="text"
              value=""
            />
          </div>
        </form>
      </div>
    </div>
    <strong
      className="chakra-text css-62oyan"
    />
    <button
      aria-label="Switch to dark mode"
      className="chakra-button css-acoyy6"
      type="button"
    >
      <svg
        aria-hidden={true}
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
          key="0"
        />
      </svg>
    </button>
  </nav>
  <div
    className="css-0"
  >
    <div
      className="css-gmuwbf"
    >
      <div
        className="css-alkm14"
      >
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Batman"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    DC Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Batman
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Bruce Wayne
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/dc-batman"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Superman"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    DC Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Superman
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Kal-El
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/dc-superman"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Flash"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    DC Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Flash
                  </h3>
                  <p
                    className="css-0"
                  >
                    Jay Garrick
                  </p>
                  <p
                    className="css-pb69ky"
                  >
                     Barry Allen, Wally West, Bart Allen
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/dc-flash"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Green Lantern"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    DC Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Green Lantern
                  </h3>
                  <p
                    className="css-0"
                  >
                    Alan Scott
                  </p>
                  <p
                    className="css-pb69ky"
                  >
                     Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/dc-green"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Green Arrow"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    DC Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Green Arrow
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Oliver Queen
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/dc-arrow"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Wonder Woman"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    DC Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Wonder Woman
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Princess Diana
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/dc-wonder"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Martian Manhunter"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    DC Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Martian Manhunter
                  </h3>
                  <p
                    className="css-0"
                  >
                    J'onn J'onzz
                  </p>
                  <p
                    className="css-pb69ky"
                  >
                    Martian Manhunter
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/dc-martian"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Robin/Nightwing"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    DC Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Robin/Nightwing
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Dick Grayson
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/dc-robin"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Blue Beetle"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    DC Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Blue Beetle
                  </h3>
                  <p
                    className="css-0"
                  >
                    Dan Garret
                  </p>
                  <p
                    className="css-pb69ky"
                  >
                     Ted Kord, Jaime Reyes
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/dc-blue"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Black Canary"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    DC Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Black Canary
                  </h3>
                  <p
                    className="css-0"
                  >
                    Dinah Drake
                  </p>
                  <p
                    className="css-pb69ky"
                  >
                     Dinah Lance
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/dc-black"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Spider Man"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    Marvel Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Spider Man
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Peter Parker
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/marvel-spider"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Captain America"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    Marvel Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Captain America
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Steve Rogers
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/marvel-captain"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Iron Man"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    Marvel Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Iron Man
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Tony Stark
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/marvel-iron"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Thor"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    Marvel Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Thor
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Thor Odinson
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/marvel-thor"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Hulk"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    Marvel Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Hulk
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Bruce Banner
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/marvel-hulk"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Wolverine"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    Marvel Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Wolverine
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    James Howlett
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/marvel-wolverine"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Daredevil"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    Marvel Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Daredevil
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Matthew Michael Murdock
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/marvel-daredevil"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Hawkeye"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    Marvel Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Hawkeye
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Clinton Francis Barton
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Cyclops"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    Marvel Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Cyclops
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Scott Summers
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/marvel-cyclops"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
        Array [
          <div
            className="chakra-offset-slide"
            style={
              Object {
                "opacity": 0,
                "transform": "scale(0.5) translateZ(0)",
              }
            }
          >
            <div
              className="css-ekpko4"
            >
              <div
                className="css-zaltu2"
              >
                <img
                  alt="Silver Surfer"
                  className="chakra-image__placeholder css-17z15b1"
                  src="https://via.placeholder.com/150"
                />
                <div
                  className="css-1booatq"
                >
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    Marvel Comics
                  </span>
                  <h3
                    className="css-jl8910"
                  >
                    Silver Surfer
                  </h3>
                  <p
                    className="css-pb69ky"
                  >
                    Norrin Radd
                  </p>
                  <a
                    className="chakra-link css-181ltmm"
                    href="/heroe/marvel-silver"
                    onClick={[Function]}
                  >
                    More
                  </a>
                </div>
              </div>
            </div>
          </div>,
          "",
          "",
        ]
      </div>
    </div>
  </div>
</div>
`;
